Show summary info

adamrp opened the pull request:
There are a lot of commits here (sorry about that, was necessary to properly test and trouble-shoot on webdev).

If someone wants to sit down with me, I can review the changes in person if that would help.

Commit:
Change study_id retrieval

study_id is now pulled from form if it is not present in sess,
allowing exort_sample_grid.psp to be used in contexts where the
study_id may not be defined in sess.

Commit:
Remove unused variable study_dir

Commit:
Add sample summary table function

This is the code that is used to display summary information about samples,
previously only available via study management tools. Because we want to
be able to display this data in other contexts, I wrote this function to
avoid code duplication.

Commit:
Add function for sample grid display

This is the code that is used to display summary information about samples,
previously only available via study management tools. Because we want to
be able to display this data in other contexts, I wrote this function to
avoid code duplication.

Commit:
Add message when study has not yet been processed

Commit:
Change code for study_id lookup

Commit:
Change select_study_task, use new grid display fn

Commit:
Merge branch 'master' of https://github.com/qiime/qiime_web_app into show_summary_info

Commit:
show split_libraries log header, test1

Commit:
Add another link to download histograms

Commit:
Add fn to convert tab-delim lines to HTML table

Commit:
Add fn to generate HTML from split_lib histograms

Commit:
Add sample grid to study summary page

Commit:
New page to export split_lib histograms download

douginator2000 commented:
Nice, thanks for factoring out the code into a new file. Portable is good. Is this deployed on webdev currently?

adamrp commented:
It is -- but note that the histograms and sample grids will all be blank
(unless we copy some finished results from www over to webdev)

On Mon, Apr 15, 2013 at 3:47 PM, Doug Wendel notifications@github.comwrote:

> Nice, thanks for factoring out the code into a new file. Portable is good.
> Is this deployed on webdev currently?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413605
> .

douginator2000 commented:
When I click on "Histograms" I get this error (selected study Arm_hand_sites
on webdev):

MOD_PYTHON ERROR

ProcessId:      13611
Interpreter:    'webdev.microbio.me'

ServerName:     'webdev.microbio.me'
DocumentRoot:   '/var/wwwroot/htdocs'

URI:            '/qiime/export_histograms.psp'
Location:       None
Directory:      '/var/wwwroot/htdocs/qiime/'
Filename:       '/var/wwwroot/htdocs/qiime/export_histograms.psp'
PathInfo:       ''

Phase:          'PythonHandler'
Handler:        'mod_python.psp'

Traceback (most recent call last):

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1537, in HandlerDispatch
    default=default_handler, arg=req, silent=hlist.silent)

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1229, in _process_target
    result = _execute_target(config, req, object, arg)

  File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
line 1128, in _execute_target
    result = object(arg)

  File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
line 337, in handler
    p.run()

  File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
line 243, in run
    exec code in global_scope

  File "/var/wwwroot/htdocs/qiime/export_histograms.psp", line 9, in <module>
    from summarize_seqs_otu_hits import histograms_as_html_table

ImportError: cannot import name histograms_as_html_table

On Mon, Apr 15, 2013 at 3:50 PM, adamrp notifications@github.com wrote:

> It is -- but note that the histograms and sample grids will all be blank
> (unless we copy some finished results from www over to webdev)
> 
> On Mon, Apr 15, 2013 at 3:47 PM, Doug Wendel notifications@github.comwrote:
> 
> > Nice, thanks for factoring out the code into a new file. Portable is
> > good.
> > Is this deployed on webdev currently?
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413605>
> > .
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413734
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

adamrp commented:
I saw that a few times, too, and I have no idea what's causing it.  The
function definitely exists (take a look at the file), but for some reason
it won't import it sometimes.  If you remove the .pyc file, it will
successfully import....  This also happens for the
utils.psp_utils.tab_delim_lines_to_table function.  Does anyone have any
ideas on that?

On Mon, Apr 15, 2013 at 3:52 PM, Doug Wendel notifications@github.comwrote:

> When I click on "Histograms" I get this error (selected study
> Arm_hand_sites
> on webdev):
> 
> MOD_PYTHON ERROR
> 
> ProcessId: 13611
> Interpreter: 'webdev.microbio.me'
> 
> ServerName: 'webdev.microbio.me'
> DocumentRoot: '/var/wwwroot/htdocs'
> 
> URI: '/qiime/export_histograms.psp'
> Location: None
> Directory: '/var/wwwroot/htdocs/qiime/'
> Filename: '/var/wwwroot/htdocs/qiime/export_histograms.psp'
> PathInfo: ''
> 
> Phase: 'PythonHandler'
> Handler: 'mod_python.psp'
> 
> Traceback (most recent call last):
> 
> File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
> line 1537, in HandlerDispatch
> default=default_handler, arg=req, silent=hlist.silent)
> 
> File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
> line 1229, in _process_target
> result = _execute_target(config, req, object, arg)
> 
> File "/usr/local/lib/python2.7/site-packages/mod_python/importer.py",
> line 1128, in _execute_target
> result = object(arg)
> 
> File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
> line 337, in handler
> p.run()
> 
> File "/usr/local/lib/python2.7/site-packages/mod_python/psp.py",
> line 243, in run
> exec code in global_scope
> 
> File "/var/wwwroot/htdocs/qiime/export_histograms.psp", line 9, in
> <module>
> from summarize_seqs_otu_hits import histograms_as_html_table
> 
> ImportError: cannot import name histograms_as_html_table
> 
> On Mon, Apr 15, 2013 at 3:50 PM, adamrp notifications@github.com wrote:
> 
> > It is -- but note that the histograms and sample grids will all be blank
> > (unless we copy some finished results from www over to webdev)
> > 
> > On Mon, Apr 15, 2013 at 3:47 PM, Doug Wendel notifications@github.comwrote:
> > 
> > > Nice, thanks for factoring out the code into a new file. Portable is
> > > good.
> > > Is this deployed on webdev currently?
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub<
> > > https://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413605>
> > > .
> > 
> > —
> > Reply to this email directly or view it on GitHub<
> > https://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413734>
> > .
> 
> ## 
> 
> Doug Wendel
> Professional Research Associate
> University of Colorado at Boulder
> Department of Chemistry and Biochemistry, Knight Lab
> 303-641-5442
> wendel@colorado.edu
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16413842
> .

ElDeveloper commented:
If you think that it would help we can go through the code together, just let me know when you want to work with this.

adamrp commented:
Sure!  Tomorrow between 12:30 and 1 could work, or early (like 9)

On Mon, Apr 15, 2013 at 4:00 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> If you think that it would help we can go through the code together, just
> let me know when you want to work with this.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16414249
> .

adamrp commented:
apachectl restart seemed to the trick... the code is still deployed on webdev right now, let me know if you see any errors like that again...

Commit:
Skip blank lines

Commit:
Add tests for tab_delim_lines_to_table

ElDeveloper commented on python_code/study_summary.py:
```diff
@@ -105,3 +105,72 @@ def print_study_info_and_values_table(query_results, data_access):
     return ''.join(info_table)
     
+def get_sample_summary_html(study_id, data_access):
+    out_string = ''
```
:shipit: docstring?

ElDeveloper commented on python_code/summarize_seqs_otu_hits.py:
```diff
@@ -122,6 +123,12 @@ def summarize_all_stats(processed_dir):
     return processed_results
 
+def histograms_as_html_table(processed_dir):
+    """
```
:shipit: docstring.

ElDeveloper commented on python_code/summarize_seqs_otu_hits.py:
```diff
@@ -122,6 +123,12 @@ def summarize_all_stats(processed_dir):
+    """
+    """
+    histograms_fp = join(processed_dir, 'split_libraries', 'histograms.txt')
+    return tab_delim_lines_to_table(open(histograms_fp).readlines())
```
U

ElDeveloper commented on python_code/tests/test_psp_utils.py:
```diff
@@ -37,5 +38,84 @@ def test_format_submit_form_to_fusebox_string(self):
         self.assertEqual(format_submit_form_to_fusebox_string(
             page='portal.psp'), change_page_only)
 
+    def test_tab_delim_lines_to_table(self):
```
docstring

ElDeveloper commented on python_code/utils/psp_utils.py:
```diff
@@ -25,3 +25,29 @@ def format_submit_form_to_fusebox_string(**kwargs):
+    table_str = '<table ' + table_opts + '>\n'
+
+    for line in lines:
+        if line and line[0] == '#':
```
use startswith instead

ElDeveloper commented on www/export_histograms.psp:
```diff
@@ -0,0 +1,37 @@
+sess = Session.Session(req)
+data_access = data_access_factory(ServerConfig.data_access_type)
+
+study_id = sess.get('study_id', form.get('study_id', ''))
```
a comment?

ElDeveloper commented on python_code/study_summary.py:
```diff
@@ -105,3 +105,72 @@ def print_study_info_and_values_table(query_results, data_access):
+        if sample_details[0][5] == None:
+            write_seq_otu_cols = False
+    else:
+        return 'This study has not yet been processed.'
```
Different message here? Or no message? Suggestions @douginator2000 ?

Commit:
Add docstring

Commit:
Add docstring; change open from default mode to U

Commit:
Add docstring

Commit:
Change HTML string generation method; change test

Instead of building up the string by successive string concatenations,
instead build up a list and then use string.join to build the final string

Commit:
Add comments explaining how study_id is retrieved

Code was slightly unclear, comment should help!

douginator2000 commented:
The way the sample grid is handled on the study page is that if the study has not been processed, it still lists everything but the counts. So if you wanted to mirror the functionality then you can display that information.

douginator2000 commented:
And to clarify, it just omits the columns that would normally display the counts.

adamrp commented:
Might be able to implement that, I'll look into that when I get a chance
later this week.

Adam
On Apr 16, 2013 2:51 PM, "Doug Wendel" notifications@github.com wrote:

> And to clarify, it just omits the columns that would normally display the
> counts.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/350#issuecomment-16470950
> .

adamrp commented:
It looks like it still displays everything, unless I'm misunderstanding something (see Agler_butyrate_from_corn_fiber, study id 741).

Commit:
Change message when no metadata is available

adamrp commented:
This should be ready to go, unless there are further comments on the code.

ElDeveloper commented:
This looks good. Mr. Señor Developer (a. k. a. @douginator2000) would you mind giving this a last check? Adam and I went through everything but a last check would be good :shipit:

douginator2000 commented:
Looks good to me!
